docs/source/examples/tips_and_tricks.rst

```diff
@@ -0,0 +1,492 @@
+
+        mask = mask_padding_sequences(lengths)
+        masked_target = torch.where(mask, padded_sequences_labels,
+                                    torch.ones_like(mask) * mask_value)
+
+        # We also pass the mask for the F1 score since it need a mask tensor to be compute
+        return (padded_sequences_vectors, lengths), (masked_target, mask)
+
+     def mask_padding_sequences(lengths):
+        """
+        Create a mask from the lengths of the padded sequences.
+
+        Args:
+            lengths: The lengths use to create the padded sequence.
+        """
+
+        max_len = lengths[0]
+        mask = torch.arange(max_len).expand(len(lengths), max_len) < lengths.unsqueeze(1)
+        return mask.bool()
+
+.. code-block:: python
+
+    train_loader = DataLoader(train_data_vectorize, batch_size=batch_size, shuffle=True, collate_fn=PadCollate())
+    valid_loader = DataLoader(valid_data_vectorize, batch_size=batch_size, collate_fn=PadCollate())
+    test_loader = DataLoader(test_data_vectorize, batch_size=batch_size, collate_fn=PadCollate())
+
+Full Network
+^^^^^^^^^^^^
+
+Now, since we have packed the sequence, we cannot use the PyTorch ``nn.Sequential`` constructor to define our model, so we will define the forward pass for it to unpack the sequences.
+
+.. code-block:: python
+
+    class FullNetWork(nn.Module):
+        def __init__(self, lstm_network, fully_connected_network):
+            super().__init__()
+            self.hidden_state = None
+
+            self.lstm_network = lstm_network
+            self.fully_connected_network = fully_connected_network
+
+        def forward(self, padded_sequences_vectors, lengths):
+            """
+                Defines the computation performed at every call.
+            """
+            pack_padded_sequences_vectors = pack_padded_sequence(padded_sequences_vectors, lengths, batch_first=True)
+
+            lstm_out, self.hidden_state = self.lstm_network(pack_padded_sequences_vectors)
+            lstm_out, _ = pad_packed_sequence(lstm_out, batch_first=True)
+
+            tag_space = self.fully_connected_network(lstm_out)
+            return tag_space.transpose(-1, 1) # we need to transpose since it's a sequence
+
+    full_network = FullNetWork(lstm_network, fully_connected_network)
+
+Summary
+-------
+
+So we have created an LSTM network (``lstm_network``), a fully connected network (``fully_connected_network``), those two components are used in the full network. This full network used padded, packed sequences (defined in the forward pass), so we created the ``PadCollate`` class to process the need work. The DataLoader will conduct that process. Finally, when we load the data, this will be done using the vectorizer, so the address will be represented using word embeddings. Also, the address components will be converted into categorical value (from 0 to 7).
+
+The Training Loop
+=================
```

Comment:
I think that we should not put this section title and put the sections below it as first-level section title instead so that they appear in the left menu since it's what we want to show.